src/sentry/workflow_engine/processors/delayed_workflow.py

```diff
@@ -790,6 +976,70 @@ def fire_actions_for_groups(
     )
 
 
+@sentry_sdk.trace
+def fire_actions_for_groups_v2(
```

Comment:
This function is greatly simplified from the original to only fire the actions associated with the passing IF DCGs, which should be the only type of DCG within `groups_to_fire`